spec/models/coronavirus/announcement_spec.rb

```diff
@@ -82,4 +91,29 @@
       expect(page.announcements.count).to eq 1
     end
   end
+
+  describe "#published_at=" do
+    let(:announcement) { build(:coronavirus_announcement) }
+
+    it "can accept published_at as a hash" do
+      announcement.published_at = { "day" => "1", "month" => "1", "year" => "2020" }
+      expect(announcement.published_at).to eq(Time.zone.local(2020, 1, 1))
+    end
+
+    it "sets published_at to nil for an invalid date" do
+      announcement.published_at = { "day" => "1", "month" => "13", "year" => "2020" }
+      expect(announcement.published_at).to be_nil
+    end
+
+    it "sets published_at to nil for an empty date" do
+      announcement.published_at = { "day" => "", "month" => "", "year" => "" }
+      expect(announcement.published_at).to be_nil
+    end
+
+    it "can still accept published_at as a time" do
```

Comment:
I'm not sure why we need this test?

Reply:
I put this in as it's the only unit test that executes the https://github.com/alphagov/collections-publisher/pull/1335/files#diff-96d56b8391c8046c6b0d653114f673cfddced5572fe43faeecf16c959fe2b769R12-R16 path of code. I'm not sure if it's tested implicitly elsewhere in the codebase though.